ttnn/cpp/ttnn/operations/normalization/layernorm/device/layernorm_op.cpp

```diff
@@ -131,7 +154,20 @@ void LayerNorm::validate(const std::vector<Tensor> &input_tensors, const std::ve
 }
 std::vector<tt::tt_metal::Shape> LayerNorm::compute_output_shapes(const std::vector<Tensor> &input_tensors) const {
     const auto& input_tensor = input_tensors.at(0);
-    return {input_tensor.get_legacy_shape()};
+    if (this->distributed_norm_stage == DistributedLayerNormStage::PRE_ALL_GATHER)
+    {
+        auto output_shape = input_tensor.get_legacy_shape();
+        auto padding = output_shape.padding();
+        uint32_t num_tiles_w = this->norm_type == LayerNormType::LAYERNORM ? 2 : 1;
+        output_shape[3] = num_tiles_w * TILE_WIDTH;
+        padding[3] = Padding::PadDimension{0, 31};
```

Comment:
Stats is one tile in case of rms_norm. 
Only 1st column of the tile contains data(E[x^2]), rest is padding.
Actually, padding is incorrect, will fix it